Sources/ProjectDescription/WorkspaceGenerationOptions.swift

```diff
@@ -0,0 +1,54 @@
+extension Workspace {
+    /// Contains options related to the workspace generation.
+    public struct GenerationOptions: Codable, Equatable {
+        /// Represents the behavior Xcode will apply to the workspace regarding
+        /// schema generation using the `IDEWorkspaceSharedSettings_AutocreateContextsIfNeeded` key.
+        /// - seealso: `WorkspaceSettingsDescriptor`
+        public enum AutomaticXcodeSchemes: String, Codable, Equatable {
+            /// Will not add the key to the settings file.
+            case `default`
+
+            /// Will add the key with the value set to `false`.
+            case disabled
+
+            /// Will add the key with the value set to `true`.
+            case enabled
+        }
```

Comment:
Why not just use a `Bool?` for this?

Reply:
Yes, I'll change it in another PR. This is just a 1-1 mapping of the old one

Reply:
The reasoning for this was leaving the Xcode defaults when set to `default` _(i.e. don't bother generating some of the unnecessary files if you don't need to tweak the default behaviour)_. 

I think it's fine as is, if it were to change would probably need to be an optional bool to support those 3 states. 

https://github.com/tuist/tuist/pull/3832

Reply:
Ah my bad misread!

> Why not just use a `Bool?` for this?

That's exactly what you suggested 😅

Reply:
changed in this PR as it was a small change, and also changed default to false as discussed in today's get-together